### PR TITLE
Performance Profile

### PR DESCRIPTION
**Pull request by tangledbytes**
**Description**

This PR adds Performance Profile feature in meshery. Following are the APIs added in the meshery server:
1. CRUD for Performance profiles
2. CRUD for schedules (this is not yet exposed in the UI)
3. CRUD for performance profiles test results

Following is a short demo of performance profle UI:

perf-profile

**Notes for Reviewers**
APIs for schedules are in place but they aren't yet exposed in the UI. So the "next run" column in the profiles table is missing (code is commented) and form for the same is also missing.

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Well-done demo and recording of this feature, @utkarsh-pro. This user experience is sophisticated, intriguing, involved, complex, powerful. Additional thought needs to be given as to how to simplify the interaction, increasing its intuitiveness. The current design isn't far from achieving that. A verbal walk-through will be the most helpful in explore what tweaks will get it across the line.

File: router/server.go [hunk 1]
```diff
@@ -133,6 +133,15 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 		Methods("DELETE")
 	gMux.HandleFunc("/api/experimental/oam/{type}", h.OAMRegisterHandler).Methods("GET", "POST")
 
+	gMux.Handle("/api/user/performance/profiles", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.GetPerformanceProfilesHandler)))).
```
Comment (leecalcote) [inline router/server.go:132]: @dhruv0000 more apis to add to the endpoint spreadsheet.

**Comment by Nikhil-Ladha** [comment]
Finally, the profiles are here.....looks so good @utkarsh-pro 👌🏼

**Comment by utkarsh-pro** [comment]
Closing this in favour of #2549

**Pull request closed without merging**